config/src/main/java/org/axonframework/config/AggregateConfigurer.java

```diff
@@ -178,7 +182,13 @@ protected AggregateConfigurer(Class<A> aggregate) {
         );
         snapshotTriggerDefinition = new Component<>(() -> parent, name("snapshotTriggerDefinition"),
                                                     c -> NoSnapshotTriggerDefinition.INSTANCE);
-        snapshotFilter = new Component<>(() -> parent, name("snapshotFilter"), c -> SnapshotFilter.allowAll());
+        snapshotFilter = new Component<>(() -> parent, name("snapshotFilter"), c -> {
+            Optional<String> revisionValue =
+                    AnnotationUtils.findAnnotationAttribute(aggregate, Revision.class, "revision");
+            return revisionValue.isPresent()
+                    ? new RevisionSnapshotFilter(revisionValue.get())
```

Comment:
Wouldn't this change break backwards compatibility? I'd opt for finding a way to toggle this feature instead of using it by default.

Reply:
There was no reason to put the `@Revision` on an Aggregate until now. The only thing is will "break", is that now the default is to look at the revision of the aggregate to validate whether a snapshot is suitable for that Aggregate. If that alters any behavior, it will cause an aggregate to be streamed from its entire past events for a single time, whereafter a new snapshot is created.

Reply:
Ah, it wasn't used. Please disregard my comment then :)